salt/modules/yumpkg.py

```diff
@@ -1957,7 +1965,24 @@ def remove(name=None, pkgs=None, **kwargs):  # pylint: disable=W0613
         raise CommandExecutionError(exc)
 
     old = list_pkgs()
-    targets = [x for x in pkg_params if x in old]
+    targets = []
+    for target in pkg_params:
+        # Check if package version set to be removed is actually installed:
+        # old[target] contains a comma-separated list of installed versions
+        if target in old and not pkg_params[target]:
+            targets.append(target)
+        elif target in old and pkg_params[target] in old[target].split(','):
```

Comment:
call pkgs_list with versions_as_list=True    then no need to split

Reply:
If I add `versions_as_list=True` then it would also affect to the structure of returning `new` and `old` attributes of the `changes` dict as it would be a list of string instead of a comma-separated string. I didn't want to change that behavior here.